e2e/nomostest/sync-status-watch-controller.go

```diff
@@ -0,0 +1,115 @@
+const StatusWatchName = "sync-status-watch"
+
+// ImagePlaceholder is the placeholder in the manifest that will be replaced with the actual image.
+const ImagePlaceholder = "SYNC_STATUS_WATCH_CONTROLLER_IMAGE"
+
+// ManifestPath is the path to the manifest file.
+const ManifestPath = "../../examples/post-sync/sync-watch-manifest.yaml"
+
+// TestImage is the image used for testing the sync status watch controller.
+const TestImage = testing.TestInfraArtifactRepositoryAddress + "/sync-status-watch-controller:v1.0.0-efef3e1d"
+
+// SetupSyncStatusWatchController sets up the sync status watch controller in the cluster.
+// It creates the necessary namespace and deploys the controller.
+func SetupSyncStatusWatchController(nt *NT) error {
+	nt.T.Logf("applying sync status watch controller manifest")
+
+	// Create namespace first to ensure it exists
+	namespace := testSyncStatusWatchNamespace()
+	if err := nt.KubeClient.Create(namespace); err != nil && !apierrors.IsAlreadyExists(err) {
+		return fmt.Errorf("creating namespace %s: %w", namespace.Name, err)
+	}
+
+	if err := execManifestCommand(nt, "apply"); err != nil {
+		nt.describeNotRunningTestPods(StatusWatchNamespace)
+		return fmt.Errorf("applying sync status watch manifest: %w", err)
+	}
+
+	return nt.Watcher.WatchForCurrentStatus(kinds.Deployment(), StatusWatchName, StatusWatchNamespace)
+}
+
+// TeardownSyncStatusWatchController tears down the sync status watch controller from the cluster.
+// It removes the namespace and associated resources.
+func TeardownSyncStatusWatchController(nt *NT) error {
+	nt.T.Log("tearing down sync status watch controller")
+
+	if err := execManifestCommand(nt, "delete"); err != nil {
+		nt.T.Logf("Error deleting resources: %v", err)
+	}
+
+	if err := nt.KubeClient.Delete(testSyncStatusWatchNamespace()); err != nil && !apierrors.IsNotFound(err) {
+		nt.T.Logf("Error deleting Namespace: %v", err)
+	}
+
+	nt.T.Log("Sync status watch controller teardown complete")
+	return nil
+}
+
+// execManifestCommand executes a kubectl command to apply or delete a manifest with the test image.
+func execManifestCommand(nt *NT, kubectlAction string) error {
+	input, err := os.ReadFile(ManifestPath)
+	if err != nil {
+		return fmt.Errorf("reading manifest file %s: %w", ManifestPath, err)
+	}
+
+	content := strings.ReplaceAll(string(input), ImagePlaceholder, TestImage)
+
+	tmpFile, err := os.CreateTemp("", "statuswatch-*.yaml")
+	if err != nil {
+		return fmt.Errorf("creating temp file: %w", err)
+	}
+	defer func() {
+		if removeErr := os.Remove(tmpFile.Name()); removeErr != nil {
+			nt.T.Logf("Warning: failed to remove temporary file %s: %v", tmpFile.Name(), removeErr)
+		}
+	}() // Clean up the temp file when done
+
+	if err := os.WriteFile(tmpFile.Name(), []byte(content), 0644); err != nil {
+		return fmt.Errorf("writing to temp file: %w", err)
+	}
+
+	_, err = nt.Shell.ExecWithDebug("kubectl", kubectlAction, "-f", tmpFile.Name())
```

Comment:
nit: I guess kubectl apply is ok, but I would have probably parsed the manifest and used the KubeClient, since it gives back better errors and avoids needing a temp file. Then again, using kubectl is more likely to be what customers will do... unless they deploy it with CS.